### ISSUE TITLE
a question about computational formula in gee code

### ISSUE THREAD

**Issue by Ke-Albert** [ChenY04/GEE#4]
Dear Professor Chen,
I got stuck when I analyzed your GF-SG gee code in the newest version, and I compared this formula in your paper, I found some differences about this formula between the code and your published paper. The formula is explained in your paper as follows:
![image](https://user-images.githubusercontent.com/65125401/221812076-e8ee10ea-9726-4a20-9846-e2a4ab871983.png)
![image](https://user-images.githubusercontent.com/65125401/221812213-2daaad30-0050-415f-aa8a-c71495f4bdf1.png)
but, in gee code, the formula seems to ignore computing the 'Rj(x,y)' but use 'corj(x,y)' directly. The recipe in gee is as follows:
'''
//Get neighboring similar pixels with the correlation coefficient above 0.8
var threshold_img = ee.Image.constant(0.8);
var threshold_mask = modnei_landsat_corre.gte(threshold_img);
var modnei_landsat_corre_valid = modnei_landsat_corre.multiply(threshold_mask);
var modnei_landsat_corre_sum = modnei_landsat_corre_valid.reduce(ee.Reducer.sum());
var modnei_landsat_corre_normal = modnei_landsat_corre_valid.divide(modnei_landsat_corre_sum);
var modndviPair_match0 = modndviPair_nei0.map(function(img){
  var onemodis_w = img.multiply(modnei_landsat_corre_normal)
  var onemodis_normal = onemodis_w.reduce(ee.Reducer.sum());
  return onemodis_normal;});
'''
Does this is to simplify the computation by purpose?

**Comment by ChenY04** [comment]
yeah, we think the two formulas have similar results and  simplify the computation on GEE. thanks for your attention
________________________________
From: Ke-Albert ***@***.***>
Sent: Tuesday, February 28, 2023 9:47 AM
To: ChenY04/GEE ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [ChenY04/GEE] a question about computational formula in gee code (Issue #4)


Dear Professor Chen,
I got stuck when I analyzed your GF-SG gee code in the newest version, and I compared this formula in your paper, I found some differences about this formula between the code and your published paper. The formula is explained in your paper as follows:
[image]<https://user-images.githubusercontent.com/65125401/221812076-e8ee10ea-9726-4a20-9846-e2a4ab871983.png>
[image]<https://user-images.githubusercontent.com/65125401/221812213-2daaad30-0050-415f-aa8a-c71495f4bdf1.png>
but, in gee code, the formula seems to ignore computing the 'Rj(x,y)' but use 'corj(x,y)' directly. The recipe in gee is as follows:
'''
//Get neighboring similar pixels with the correlation coefficient above 0.8
var threshold_img = ee.Image.constant(0.8);
var threshold_mask = modnei_landsat_corre.gte(threshold_img);
var modnei_landsat_corre_valid = modnei_landsat_corre.multiply(threshold_mask);
var modnei_landsat_corre_sum = modnei_landsat_corre_valid.reduce(ee.Reducer.sum());
var modnei_landsat_corre_normal = modnei_landsat_corre_valid.divide(modnei_landsat_corre_sum);
var modndviPair_match0 = modndviPair_nei0.map(function(img){
var onemodis_w = img.multiply(modnei_landsat_corre_normal)
var onemodis_normal = onemodis_w.reduce(ee.Reducer.sum());
return onemodis_normal;});
'''
Does this is to simplify the computation by purpose?
